proposals/0000-stream-receipts.md

```diff
@@ -0,0 +1,93 @@
+
+## Motivation
+
+Interledger payment verification is necessary for multiple use cases including Web Monetization, where ILP payments must be verified to securely serve exclusive content, and API micropayments.
+
+Standardized receipt functionality at the STREAM layer enables the payment recipient or a third party to verify payment without having access to the corresponding ILP packets.
+
+## Conventions and Definitions
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in BCP 14 [RFC2119](https://tools.ietf.org/html/rfc2119) [RFC8174](https://tools.ietf.org/html/rfc8174) when, and only when, they appear in all capitals, as shown here.
+
+Definitions of terms that are used in this document:
+
+- **Receiver** - The party that is receiving funds over STREAM
+- **Verifier** - The party that wishes to verify that ILP payments occurred to the Receiver
+- **Sender** - The party that is sending funds over STREAM
+- **Receipt Nonce** - A random nonce used to identify a STREAM connection in a Receipt
+- **Receipt Secret** - The key used to generate a Receipt's HMAC
+- **Receipt** - A binary object containing a version number, the Receipt Nonce, a Stream ID on the STREAM connection, an integer amount of funds received on a stream, and an HMAC over all those fields
+
+## Overview
+
+In a STREAM connection, a Receipt is generated by the Receiver for every fulfilled ILP packet. Each Receipt on a stream contains a progressively higher amount, representing the total amount received on that stream. So the latest Receipt on a stream replaces any previous Receipts on that stream.
+
+The Verifier SHOULD consider a Receipt valid if:
+
+- The Receipt contents produce an HMAC (using the Receipt Secret as the seed) identical to the Receipt's HMAC field.
+- The amount is greater than any other Receipts with that Receipt Nonce and Stream ID.
+- The Receipt is not stale.
+
+The Verifier SHOULD determine a length of time starting from when a Receipt Nonce is generated that corresponding Receipts will be considered valid. Mechanisms by which the Verifier can check Receipts' staleness include, but are not limited to:
+- persistently storing the Receipt Nonce and a timestamp relative to when the Nonce was generated
+- encoding a timestamp in the Receipt Nonce
+
+If keeping a balance of how much the Sender has paid, the Verifier SHOULD store the Receipt amount under the Receipt Nonce + Stream ID. If there is already an amount stored for that Stream ID, the Verifier SHOULD only credit the Sender the difference in Receipt amounts.
+
+Alteratively, the Verifier MAY forego checking the Receipt amount if they are only concerned with whether or not the Sender paid at all, as opposed to how much.
+
+### Flow
+
+This flow occurs for each STREAM connection with Receipts enabled.
+
+1. Verifier MUST generate a random Receipt Nonce.
```

Comment:
this `MUST` contradicts with the suggestion above to include the timestamp in the receipt nonce. `MUST generate a unique Receipt Nonce` would be more accurate